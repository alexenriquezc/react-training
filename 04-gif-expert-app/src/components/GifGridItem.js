import React from 'react'

export const GifGridItem = ({ title, url }) => {
    const rand = Math.random();
    return (
        <div className="card animate__animated animate__fadein" key={rand}>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
