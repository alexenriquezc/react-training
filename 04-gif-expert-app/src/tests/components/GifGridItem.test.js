import { GifGridItem } from "../../components/GifGridItem"
import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';

describe('Testing GifGridItem Componetn', () => {
    const title = 'test';
    const url = 'http://localhost'; 
    const wrapper = shallow(<GifGridItem url={url} title={title}/>);
    
    test('should displays component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should has a paragraph with the title', () => {
        const paragraph = wrapper.find('p');
        expect( paragraph.text().trim()).toBe(title);
    });

    test('should  obtains img equals to url and alt', () => {
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('should has animate__fadeIn', () => {
        const div = wrapper.find('div');
        const hasClass = div.hasClass('animate__fadeIn');
        expect(hasClass).toBe( true );

        // const className = div.prop('className');
        // expect(className.includes('animate__fadeIn')).toBe( true );
    })
    
    
    
})
