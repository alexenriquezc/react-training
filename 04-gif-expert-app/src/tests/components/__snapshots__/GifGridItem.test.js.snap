// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem Componetn should displays component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test"
    src="http://localhost"
  />
  <p>
    test
  </p>
</div>
`;
